src/scheduler/analysis.py

```diff
@@ -528,3 +640,269 @@ def _fix_system_version(system_version: str | None) -> str:
+        #: Here it is just the whole FileObject
+        # We need this because the scheduler is using multiple processes which
+        # communicate via multiprocessing.Queue's.
+        # Our implementation has no "master" process which contains all the
+        # state but rather the state is passed through the queues,
+        # even if a process (like PluginRunner) does not need all state (e.g.
+        # FileObject.scheduled_analysis)
+        scheduler_state: FileObject
+
+        class Config:
+            arbitrary_types_allowed = True
+
+    def __init__(
+        self,
+        plugin: analysis.PluginV0,
+        config: Config,
+        schemata: typing.Dict[str, pydantic.BaseModel],
+    ):
+        # mp.Queue[..] works: https://github.com/python/cpython/pull/19423
+        # pylint: disable=unsubscriptable-object
+        self._plugin = plugin
+        self._config = config
+        self._schemata = schemata
+
+        self._in_queue: mp.Queue[PluginRunner.Task] = mp.Queue()
+        #: Workers put the ``Task.scheduler_state`` and the finished analysis in the out_queue
+        self.out_queue: mp.Queue[tuple[FileObject, dict]] = mp.Queue()
+        self.out_queue.close()
+
+        self.stats = mp.Array(ctypes.c_float, ANALYSIS_STATS_LIMIT)
+        self.stats_count = mp.Value('i', 0)
+        self._stats_idx = mp.Value('i', 0)
+
+        self._fsorganizer = FSOrganizer()
+
+        worker_config = Worker.Config(
+            timeout=self._config.timeout,
+        )
+        self._workers = [
+            Worker(
+                plugin=plugin,
+                worker_config=worker_config,
+                in_queue=self._in_queue,
+                out_queue=self.out_queue,
+                stats=self.stats,
+                stats_count=self.stats_count,
+                stats_idx=self._stats_idx,
+            )
+            for _ in range(self._config.process_count)
+        ]
+
+    def get_queue_len(self) -> int:
+        return self._in_queue.qsize()
+
+    def get_active_worker_count(self) -> int:
+        """Returns the amount of workers that currently analyze a file"""
+        return sum([worker.is_alive() for worker in self._workers])
+
+    def start(self):
+        for worker in self._workers:
+            worker.start()
+
+    def shutdown(self):
+        for worker in self._workers:
+            worker.terminate()
+
+    def queue_analysis(self, file_object: FileObject):
+        """Queues the analysis of ``file_object`` with ``self._plugin``.
+        The caller of this method has to ensure that the dependencies are fulfilled.
+        """
+        dependencies = {}
+        for dependency in self._plugin.metadata.dependencies:
+            Schema = self._schemata[dependency]
+            # Try to convert to the schema defined by the plugin
+            result = file_object.processed_analysis[dependency]['result']
+            dependencies[dependency] = Schema(**result)
```

Comment:
This shouldn't be a problem since the scheduler inserts the plugins in an order that avoids this plugin.

Also: Document that plugins always depend on the latest version.